ultravox/data/datasets.py

```diff
@@ -275,13 +301,14 @@ class VoiceDataset(abc.ABC, data.IterableDataset):
     BASE_AUDIO_COLUMNS = ["audio"]
 
     def __init__(self, args: VoiceDatasetArgs) -> None:
-        super().__init__()
+        abc.ABC.__init__(self)
         self._args = args
         self._session: Optional[requests.Session] = None
         self._rng = np.random.default_rng(self._args.shuffle_seed)
         self._weight = 1.0  # the default weight for the dataset
 
-    def _init_dataset(self, dataset: data.Dataset) -> None:
+    def _init_dataset(self, dataset: SizedIterableDataset) -> None:
+        SizedIterableDataset.__init__(self, dataset, len(dataset))
```

Comment:
If dataset is already a `SizedInterableDataset`, why `__init__` is needed here?

Reply:
this is so we can use the len() and iter function from the sizeiterabledataset. if we dont init, we have to implement those methods ourselves in this class.